main-command/src/main/scala/sbt/BasicCommands.scala

```diff
@@ -87,10 +87,20 @@ object BasicCommands {
   }
 
   def runHelp(s: State, h: Help)(arg: Option[String]): State = {
-    val message = try Help.message(h, arg)
+
+    val (extraArgs, remainingCommands) = s.remainingCommands match {
```

Comment:
Why is it necessary to pop a remaining command from the queue?

Reply:
Never mind you explained it above:

> Made help command "consume" commands that go after it so when user types `sbt help new` it behaves similar to `sbt "help new"`.

Reply:
Would there be a situation where the user might need more than one argument?

Reply:
What's the behaviour if you run `sbt help new compile`?

Reply:
It will print `No matches for regular expression 'new compile'.` Since `help` is kind of interactive command, I'm assuming that user wouldn't need to supply other commands after it.